workflow/controller/operator.go

```diff
@@ -2587,6 +2601,24 @@ func (woc *wfOperationCtx) addChildNode(parent string, child string) {
 	woc.updated = true
 }
 
+// getFailHosts returns slice of all child nodes with fail or error status
+func (woc *wfOperationCtx) getFailHosts(parent string) []string {
+	parentID := woc.wf.NodeID(parent)
+	parentNode, ok := woc.wf.Status.Nodes[parentID]
+	if !ok {
+		panic(fmt.Sprintf("parent node %s not initialized", parent))
```

Comment:
please don't panic - return nil or error instead

Reply:
Panic is removed now.